manifests/charts/base/templates/zzz_profile.yaml

```diff
@@ -15,8 +15,11 @@ However, we can workaround this by placing all of (1) under a specific key (.Val
 We can then merge the profile onto the defaults, then the user settings onto that.
 Finally, we can set all of that under .Values so the chart behaves without awareness.
 */}}
-{{- $defaults := $.Values.defaults }}
-{{- $_ := unset $.Values "defaults" }}
+{{- $defaults := ($.Files.Get "files/default-values.gen.yaml" | fromYaml) }}
```

Comment:
Yeah - I like this better than live-munging the values scope to un-nest `default`s

Reply:
Oops... I realized I forgot the problem with this approach.

We want: `defaults < profile < user values`.

But now we have `defaults`, `profile`, and `defaults + user values`.

We would need to do some sort of deep diff to get just the user values...

Reply:
one idea is to make the profiles not just opaque yaml we merge in but rather a lost of setters 
then for each, we go and compare "defaults" vs Values and set it only if they are equal. 

That actually still doesn't work since then a user cannot override a default value back to its default if it's modified by a profile...

Reply:
> But now we have defaults, profile, and defaults + user values.

Can't we just reintroduce `defaults` (or maybe `istio-defaults`) as the top-level key in `defaults.gen.yaml`? We can insert it during the gen if we want.

Then we can merge with any precedence we want.

Reply:
We could selectively apply `defaults` if `$.Values.xx` is unset with templating funcs. That gets a little tricky tho, especially with nested keys.

Yeah - `profiles` are a bit undercooked in Helm in general. Helm takes care of the `default` + `user` merging for you, but to use profiles the way we want to use them we have to interfere with that precedence - which there is no ability to do.

Reply:
...that of course means you have to inline the yaml in docs, which has already been litigated.

So there are 3 options AFAICT:


- Keep what we have for this, it's imperfect but short of upstream Helm PRs to flesh out the profile feature (which we are probably the heaviest users of) the best we can do with this pattern.

- Go back to the "dumb basic Helm way" of `-f` precedence, which will work.

- Use wrapper charts as analogues for "values profiles" (which is what they are, really, and all they need to be for us) - I am not ready to suggest we do this because frankly our basic chart compositional hygiene isn't good enough yet.

Reply:
I am leaning towards keeping the current, but adding a clear comment on the default part to not set it. Not the best but maybe ill do the trick"

I was going to write "an error if you set defaults" but we cannot do that for the same reasons... unless we do the defaults.yaml.gen and compare that to .defaults but that seems a lot of effort for little gain